src/main/java/org/jenkinsci/plugins/docker/commons/impl/RegistryKeyMaterialFactory.java

```diff
@@ -77,21 +82,18 @@ public KeyMaterial materialize() throws IOException, InterruptedException {
 
         // read the existing docker config file, which might hold some important settings (e.b. proxies)
         FilePath configJsonPath = FilePath.getHomeDirectory(this.launcher.getChannel()).child(".docker").child(DOCKER_CONFIG_FILENAME);
-        if (configJsonPath.exists()) {
-            String configJson = configJsonPath.readToString();
-            if (StringUtils.isNotBlank(configJson)) {
-                launcher.getListener().getLogger().print("Using the existing docker config file.");
-
-                JSONObject json = JSONObject.fromObject(configJson);
-                for (String property : BLACKLISTED_PROPERTIES) {
-                    Object value = json.remove(property);
-                    if (value != null) {
-                        launcher.getListener().getLogger().print("Removing blacklisted property: " + property);
-                    }
-                }
-
-                dockerConfig.child(DOCKER_CONFIG_FILENAME).write(json.toString(), StandardCharsets.UTF_8.name());
-            }
+        dockerConfig = UpdateDockerConfigFromSource(dockerConfig, configJsonPath, BLACKLISTED_PROPERTIES);
+
+        // Read the existing docker config file from a nested config block, will probably hold some previous credentials
+        String existingDockerSecretConfigPath = this.env.get("DOCKER_CONFIG");
+        if (StringUtils.isNotBlank(existingDockerSecretConfigPath)) {
```

Comment:
This implementation would lead to two file write operations which could be avoided IMHO

Reply:
refactored to only do a single write.